sdk/tests/conformance/extensions/khr-parallel-shader-compile.html

```diff
@@ -58,36 +95,131 @@
-    if (status !== true && status !== false) testFailed("status should be a boolean");
+let extraCode = '';
 
-    gl.linkProgram(program);
-    status = gl.getProgramParameter(program, COMPLETION_STATUS_KHR);
-    if (status !== true && status !== false) testFailed("status should be a boolean");
-
-    debug("Checking that status is true when context is lost.");
-    if (loseContext) {
-        loseContext.loseContext();
-        status = gl.getShaderParameter(vs, COMPLETION_STATUS_KHR);
-        if (status !== true) testFailed("status should be true when context is lost");
+(async () => {
+
+    if (!ext) {
+        testPassed("No KHR_parallel_shader_compile support -- this is legal");
+    } else {
+        testPassed("Successfully enabled KHR_parallel_shader_compile extension");
+
+        shouldBe("ext.COMPLETION_STATUS_KHR", "0x91B1");
+
+        debug("Checking that status is a boolean.");
+        gl.shaderSource(vs, vertexSource());
+        gl.compileShader(vs);
+        let status = gl.getShaderParameter(vs, COMPLETION_STATUS_KHR);
+        if (status !== true && status !== false) testFailed("status should be a boolean");
+
+        gl.linkProgram(program);
         status = gl.getProgramParameter(program, COMPLETION_STATUS_KHR);
-        if (status !== true) testFailed("status should be true when context is lost");
+        if (status !== true && status !== false) testFailed("status should be a boolean");
+
+        debug("Constructing shader that takes >500ms to compile");
+        let measuredCompileDuration = 0;
+        extraCode = '\n    if (true) { result += vec4(0.0000001); }';
+        for (let i = 0; measuredCompileDuration < 500; i++) {
+            extraCode += extraCode;
+            extraCode += extraCode;
+            if (i < 4) continue;
+            gl.shaderSource(vs, vertexSource(extraCode));
+            gl.shaderSource(fs, fragmentSource(extraCode));
+            gl.compileShader(vs);
+            gl.compileShader(fs);
+            gl.linkProgram(program);
+            const start = performance.now();
+            if (!gl.getProgramParameter(program, gl.LINK_STATUS)) {
+                testFailed(`Shaders failed to compile.
+                            program: ${gl.getProgramInfoLog(program)}
+                            vs: ${gl.getShaderInfoLog(vs)}
+                            fs: ${gl.getShaderInfoLog(fs)}`);
+                break;
+            }
+            measuredCompileDuration = performance.now() - start;
+        }
+
+        debug('');
+        gl.shaderSource(vs, vertexSource(extraCode));
+        gl.shaderSource(fs, fragmentSource(extraCode));
+        gl.compileShader(vs);
+        gl.compileShader(fs);
+        gl.linkProgram(program);
+
+        let start = performance.now();
+        gl.getShaderParameter(fs, COMPLETION_STATUS_KHR);
+        gl.getShaderParameter(vs, COMPLETION_STATUS_KHR);
+        let duration = performance.now() - start;
+        if (performance.now() - start > 100)
```

Comment:
Should this read "if (duration > 100)"?

Also concerned about performance on slower machines, like Chromium's testing fleet - though I agree there's no reason these calls should take any appreciable period of time.